Add translations and flash messages

MelissaKaulfuss opened the pull request:
This pull request:
- Includes translations
- Flash alerts and notices as needed
- Tests the unhappy path when creating a user (in the `users_controller_spec.rb`)

https://trello.com/c/AyUpYcN8/47-must-have-add-translations

MelissaKaulfuss committed:
Create User views now has translations

MelissaKaulfuss committed:
Using flash messages

MelissaKaulfuss committed:
Creating a user with invalid params receives alert

MelissaKaulfuss committed:
Disable Layout/FirstParameterIndentation cop

elle commented on config/locales/en.yml:
```diff
@@ -1,33 +1,15 @@
+    new:
+      title: "create your account"
+    show:
+      welcome: "welcome to stickit <%= username %>!"
```
I believe the right syntax for this is using a `%{ }` without the angle brackets or equal sign.
Ref: http://guides.rubyonrails.org/i18n.html#passing-variables-to-translations

MelissaKaulfuss replied:
You are right! Thanks. 
Fixed in sha: d886c7f

elle commented on config/locales/responders.en.yml:
```diff
@@ -3,10 +3,10 @@ en:
+        alert: '%{resource_name} could not be updated.'
       destroy:
         notice: '%{resource_name} was successfully destroyed.'
         alert: '%{resource_name} could not be destroyed.'
```
Double quotes?

MelissaKaulfuss replied:
Sorry these were still around - not sure why Rubocop ignored these?
Fixed in sha: 7b6cf80

elle commented on spec/models/user_spec.rb:
```diff
@@ -2,11 +2,11 @@
-    it  { is_expected.to validate_presence_of(:email) }
-    it  { is_expected.to have_secure_password }
+    subject { build(:user) }
+    it { is_expected.to validate_presence_of(:username) }
```
New line between 5-6?

MelissaKaulfuss replied:
Added the line - and picked up an incorrect test, I was testing `validate_presence_of(:email)` twice, instead of the second being for `validate_uniqueness_of(:email`.

Both fixed in sha: 3fded28

elle reviewed (commented):
Add `*.DS_Store` to your `.gitignore`

I usually have in mine:
```
*.DS_Store
*.swo
*.swp
```

A few small comments. Other than that, LGTM.

After you rebase, push the updated version back here to the branch, and check if the code changed or if comments are still around.

MelissaKaulfuss committed:
Ignore .DS_store file

MelissaKaulfuss committed:
Fix layout of spec and tests uniqueness of email

MelissaKaulfuss commented:
Thanks heaps Elle 🙇‍♀️
 I have added .DS_store to the project's `.gitignore` fixed in sha:  84664e5
(I had added it to my global file as you mentioned it earlier but it doesn't seem to be working).

Comments (with relevant commit shas in relevant comments).
All looks good to me, merging now.
